typed_python/_types.cpp

```diff
@@ -656,8 +656,8 @@ PyObject *refcount(PyObject* nullValue, PyObject* args) {
             )) {
         PyErr_Format(
             PyExc_TypeError,
-            "first argument to refcount %S not a permitted Type",
-            a1
+            "first argument to refcount '%S' not a permitted Type",
+            PyObject_Str(a1)
```

Comment:
So, two points. First, this will leak a PyObject.  Pretty much any function that produces a new PyObject needs a corresponding PyDecref operation to balance it. 

Secondly, I looked at the docs and %S was the right code after all. The problem is that this function is like a C-style printf - there's no type information so it just passes the argument in direclty, and we're passing a PyObjectHolder which doesn't know how to convert itself to the underlying PyObject.  So I think the right thing to do is to put it back to `%S` and to explicitly cast `a1` to a PyObject*...

Reply:
thank you